db/bbolt.go

```diff
@@ -174,6 +162,35 @@ func (bb *BBoltHandler) GetBlockByHash(hash string) (*types.Block, error) {
 	return bb.GetBlockByHeight(blockHeight)
 }
 
+func (bb *BBoltHandler) QueryIsBlockRangeFinalizedByHeight(startHeight, endHeight uint64) ([]bool, error) {
+	if startHeight > endHeight {
+		return nil, types.ErrInvalidBlockRange
+	}
+
+	// Create result slice with size of the range
+	len := endHeight - startHeight + 1
+	results := make([]bool, len)
+
+	err := bb.db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket([]byte(blocksBucket))
+
+		// Check each height in the range
+		for i := uint64(0); i < len; i++ {
+			height := startHeight + i
+			blockExists := bucket.Get(bb.itob(height)) != nil
```

Comment:
we can probably early return here given the assumption that we only store consecutive finalized blocks. so once we found one slot doesn't exist, we can `return results` directly (b/c all elements in results were initialized to `false` initially)